### PR TITLE
Don't generate Mana from storage deposit

### PR DESCRIPTION
**Pull request by cyberphysic4l**
This PR changes the potential Mana for outputs to only include tokens in excess of the minimum storage deposit, i.e., the storage deposit does not generate Mana, which makes this serve a rent payment (in Mana) for storing the output.

This PR also includes some renaming around deposits and rent. Deposit not strictly refers to the minimum storage deposit required for the object, while base token amount or amount refers to the total amount of base tokens on the object. Rent is payed in Mana, so the storage deposit itself is not "rent", but rather a deposit, and rent is the Mana that the base tokens of the storage deposit would otherwise generate if it were not a storage deposit.

### CODE REVIEW

File: output.go [hunk 1]
```diff
@@ -35,8 +35,8 @@ type Output interface {
 	NonEphemeralObject
 	ProcessableObject
 
-	// Deposit returns the amount this Output deposits.
-	Deposit() BaseToken
+	// BaseTokenAmount the amount of base tokens held by this Output.
```
Comment (PhilippGackstatter) [inline output.go]:
```suggestion
	// BaseTokenAmount returns the amount of base tokens held by this Output.
```

File: output.go [hunk 2]
```diff
@@ -294,8 +294,8 @@ func (outputIDs OutputIDs) OrderedSet(set OutputSet) Outputs[Output] {
 }
 
 var (
-	// ErrDepositAmountMustBeGreaterThanZero returned if the deposit amount of an output is less or equal zero.
-	ErrDepositAmountMustBeGreaterThanZero = ierrors.New("deposit amount must be greater than zero")
+	// ErrAmountMustBeGreaterThanZero returned if the base token amount of an output is less or equal zero.
```
Comment (PhilippGackstatter) [inline output.go]:
```suggestion
	// ErrAmountMustBeGreaterThanZero gets returned if the base token amount of an output is less or equal zero.
```

File: output.go [hunk 3]
```diff
@@ -294,8 +294,8 @@ func (outputIDs OutputIDs) OrderedSet(set OutputSet) Outputs[Output] {
...
+	ErrAmountMustBeGreaterThanZero = ierrors.New("base token amount must be greater than zero")
```
Comment (PhilippGackstatter) [inline output.go:298]: Nit: Make the error messages consistent. Write either "amount" or "base token amount" in both.

File: transaction_essence.go [hunk 4]
```diff
@@ -61,12 +61,12 @@ var (
 	ErrDelegationValidatorIDZeroed = ierrors.New("delegation output's validator ID is zeroed")
 	// ErrOutputsSumExceedsTotalSupply gets returned if the sum of the output deposits exceeds the total supply of tokens.
 	ErrOutputsSumExceedsTotalSupply = ierrors.New("accumulated output balance exceeds total supply")
-	// ErrOutputDepositsMoreThanTotalSupply gets returned if an output deposits more than the total supply.
-	ErrOutputDepositsMoreThanTotalSupply = ierrors.New("an output can not deposit more than the total supply")
+	// ErrOutputAmountMoreThanTotalSupply gets returned if an output base token amount is more than the total supply.
+	ErrOutputAmountMoreThanTotalSupply = ierrors.New("an output can not have base token amount more than the total supply")
```
Comment (PhilippGackstatter) [inline transaction_essence.go]:
```suggestion
	ErrOutputAmountMoreThanTotalSupply = ierrors.New("an output's base token amount cannot exceed the total supply")
```

**Review by PhilippGackstatter — APPROVED** [review]
Looks good to me!

File: rent.go [hunk 5]
```diff
@@ -45,20 +47,20 @@ type RentStructure struct {
...
+		return 0, ierrors.Wrapf(ErrVByteDepositNotCovered, "needed %d but only got %d", minDeposit, amount)
 	}
 
-	return minRent, nil
+	return minDeposit, nil
 }
 
-// MinRent returns the minimum rent to cover a given object.
-func (r *RentStructure) MinRent(object NonEphemeralObject) BaseToken {
+// MinDeposit returns the minimum rent to cover a given object.
```
Comment (muXxer) [inline rent.go]:
```suggestion
// MinDeposit returns the minimum deposit to cover a given object.
```

File: vm/stardust/vm_stardust.go [hunk 6]
```diff
@@ -305,20 +307,31 @@ func accountBlockIssuerSTVF(input *vm.ChainOutputWithCreationTime, next *iotago.
...
 	if err != nil {
 		return ierrors.Wrapf(err, "account %s stored mana calculation failed", current.AccountID)
 	}
 	manaIn -= manaStoredAccount
 
-	manaPotentialAccount, err := manaDecayProvider.PotentialManaWithDecay(current.Amount, input.CreationTime, vmParams.WorkingSet.Tx.Essence.CreationTime) // AccountInPotential
+	// AccountInPotential
+	// the storage deposit does not generate potential mana, so we only use the excess base tokens to calculate the potential mana
+	var excessBaseTokensAccount iotago.BaseToken
+	if current.Amount <= rentStructure.MinDeposit(current) {
```
Comment (muXxer) [inline vm/stardust/vm_stardust.go]: Can you please calculate the minDeposit only once?

File: vm/vm.go [hunk 7]
```diff
@@ -139,7 +139,12 @@ func TotalManaIn(manaDecayProvider *iotago.ManaDecayProvider, txCreationTime iot
 		}
 
 		// potential Mana
-		manaPotential, err := manaDecayProvider.PotentialManaWithDecay(input.Output.Deposit(), input.CreationTime, txCreationTime)
+		// the storage deposit does not generate potential mana, so we only use the excess base tokens to calculate the potential mana
+		if input.Output.BaseTokenAmount() <= rentStructure.MinDeposit(input.Output) {
+			continue
+		}
+		excessBaseTokens := input.Output.BaseTokenAmount() - rentStructure.MinDeposit(input.Output)
```
Comment (muXxer) [inline vm/vm.go]: Please only calculate the MinDeposit once.

**Review by muXxer — CHANGES_REQUESTED** [review]

File: vm/vm.go (hunk 7 above)
Comment (cyberphysic4l) [reply vm/vm.go]: done.

File: vm/stardust/vm_stardust.go (hunk 6 above)
Comment (cyberphysic4l) [reply vm/stardust/vm_stardust.go]: done.

**Review by muXxer — APPROVED** [review]

**Pull request merged**